buttercup/cogs/find.py

```diff
@@ -1,69 +1,238 @@
+                )
+            else:
+                status = "Claimed"
+
+        embed = (
+            Embed(color=color)
+            .add_field(name="Status", value=status)
+            .add_field(
+                name="OCR",
+                value="Yes" if self.submission["has_ocr_transcription"] else "No",
+            )
+            .add_field(
+                name="Archived", value="Yes" if self.submission["archived"] else "No"
+            )
+        )
+
+        if self.transcription:
+            embed.description = limit_str(self.transcription["text"], 200)
+        if self.submission.get("content_url"):
+            embed.set_image(url=self.submission["content_url"])
+        if self.submission.get("tor_url"):
+            embed.add_field(
+                name="ToR Post", value=f"[Link]({self.submission['tor_url']})"
+            )
+        if self.submission.get("url"):
+            embed.add_field(
+                name="Partner Post", value=f"[Link]({self.submission.get('url')})"
+            )
+        if tr_link:
+            embed.add_field(name="Transcription", value=tr_link)
+        subreddit = (
+            self.submission["url"].split("/")[4] if self.submission["url"] else None
+        )
+        if subreddit:
+            embed.set_author(
+                name=f"r/{subreddit}", url=f"https://reddit.com/r/{subreddit}"
+            )
+
+        return embed
+
+    def _clear(self) -> None:
+        """Reset the local scope caches for a new search."""
+        self.transcription: Union[Dict, None] = None
+        self.submission: Union[Dict, None] = None
+        self.author: Union[Dict, None] = None
+
+    def get_transcription(self) -> None:
+        """Get the target transcription from Blossom."""
+        for option in self.submission["transcription_set"]:
+            response = self.blossom_api.get_transcription(id=get_id_from_url(option))
+            logging.info(response.data[0])
+            if response.data[0]["author"] == self.submission["completed_by"]:
+                self.transcription = response.data[0]
+                return
+
+    def get_author(self) -> None:
+        """Retrieve the author from the submission or the transcription."""
+        if self.transcription:
+            source_url = self.transcription["author"]
+        else:
+            source_url = self.submission["claimed_by"]
+        self.author = self.blossom_api.get(
+            "volunteer/", params={"id": get_id_from_url(source_url)}
+        ).json()["results"][0]
+
+    def get_submission_from_url(self, reddit_url: str) -> Union[Dict, None]:
+        """
+        Try to get the submission corresponding to the given Reddit URL.
+
+        The URL can be a link to either a post on the partner sub, a post
+        on r/ToR, or it can be a transcription link.
+        """
+        # Determine what kind of link we're dealing with:
+        logging.critical(len(reddit_url.split("/")))
```

Comment:
`critical` is probably a bit much for a debug log :P

Reply:
hahaha you're right, this is a debugging call I forgot to remove.